server/src/main/java/com/cloud/api/query/QueryManagerImpl.java

```diff
@@ -1147,6 +1155,41 @@ public ListResponse<UserVmResponse> searchForUserVMs(ListVMsCmd cmd) {
         return response;
     }
 
+    @Override
+    public ListResponse<VirtualMachineResponse> findAffectedVmsForStorageScopeChange(FindAffectedVmsForStorageScopeChangeCmd cmd) {
+        Long poolId = cmd.getStorageId();
+        StoragePoolVO pool = storagePoolDao.findById(poolId);
+        if (pool == null) {
+            throw new IllegalArgumentException("Unable to find storage pool with ID: " + poolId);
+        }
+        if (!pool.getStatus().equals(StoragePoolStatus.Disabled)) {
+            throw new InvalidParameterValueException("Scope change of Storage pool is only allowed in Disabled state");
+        }
+        ListResponse<VirtualMachineResponse> response = new ListResponse<>();
+        if (pool.getScope() != ScopeType.ZONE) {
```

Comment:
if storage scope is  Zone, throw an exception or add a debug/info message, both are fine to me.